packages/block-editor/src/components/background-image-control/index.js

```diff
@@ -18,17 +18,19 @@ import {
 	MenuItem,
 	VisuallyHidden,
 	__experimentalItemGroup as ItemGroup,
+	__experimentalItem as Item,
 	__experimentalHStack as HStack,
 	__experimentalTruncate as Truncate,
 	Dropdown,
 	Placeholder,
 	Spinner,
+	ColorIndicator,
```

Comment:
I don't think we should use the color indicator component here. It's mainly customized for colors, and if something changes it might affect its usage here.

Also, there are some strange misshaped icons:

https://github.com/user-attachments/assets/02851c36-6314-43f2-a65a-e56814d8eaf6

I think it'd be possible to roll our own icon here using the elements that exist, and also using the changes you've made here.


https://github.com/user-attachments/assets/0b9b38f6-935a-4453-937e-fb86db9ad45a



If it helps here's the diff from my local:

<details>

<summary>Diff</summary>

```diff
diff --git a/packages/block-editor/src/components/background-image-control/index.js b/packages/block-editor/src/components/background-image-control/index.js
index 7af1888d60..3f22d45f62 100644
--- a/packages/block-editor/src/components/background-image-control/index.js
+++ b/packages/block-editor/src/components/background-image-control/index.js
@@ -24,7 +24,6 @@ import {
 	Dropdown,
 	Placeholder,
 	Spinner,
-	ColorIndicator,
 	__experimentalDropdownContentWrapper as DropdownContentWrapper,
 } from '@wordpress/components';
 import { __, _x, sprintf } from '@wordpress/i18n';
@@ -126,16 +125,22 @@ function InspectorImagePreviewItem( {
 				as="span"
 				className="block-editor-global-styles-background-panel__inspector-preview-inner"
 			>
-				<ColorIndicator
+				<span
+					className="block-editor-global-styles-background-panel__inspector-image-indicator-wrapper"
 					aria-hidden
-					style={ {
-						'--image-url': imgUrl ? `url(${ imgUrl })` : undefined,
-					} }
-					className={ clsx(
-						'block-editor-global-styles-background-panel__inspector-image-indicator',
-						{ 'has-image': !! imgUrl }
-					) }
-				/>
+				>
+					<span
+						className={ clsx(
+							'block-editor-global-styles-background-panel__inspector-image-indicator',
+							{ 'has-image': !! imgUrl }
+						) }
+						style={ {
+							'--wp-admin-background-image-preview-url': imgUrl
+								? `url(${ imgUrl })`
+								: undefined,
+						} }
+					/>
+				</span>
 				<FlexItem as="span" style={ imgUrl ? {} : { flexGrow: 1 } }>
 					<Truncate
 						numberOfLines={ 1 }
@@ -677,7 +682,7 @@ export default function BackgroundImagePanel( {
 			settings?.background?.backgroundRepeat );
 
 	return (
-		<>
+		<div className="block-editor-global-styles-background-panel__inspector-media-replace-container">
 			{ shouldShowBackgroundImageControls ? (
 				<BackgroundControlsPanel
 					label={ title }
@@ -691,9 +696,7 @@ export default function BackgroundImagePanel( {
 							style={ value }
 							inheritedValue={ resolvedInheritedValue }
 							displayInPanel
-							onResetImage={ () => {
-								resetBackground();
-							} }
+							onResetImage={ () => resetBackground() }
 							defaultValues={ defaultValues }
 						/>
 						<BackgroundSizeControls
@@ -710,11 +713,9 @@ export default function BackgroundImagePanel( {
 					style={ value }
 					inheritedValue={ resolvedInheritedValue }
 					defaultValues={ defaultValues }
-					onResetImage={ () => {
-						resetBackground();
-					} }
+					onResetImage={ () => resetBackground() }
 				/>
 			) }
-		</>
+		</div>
 	);
 }
diff --git a/packages/block-editor/src/components/background-image-control/style.scss b/packages/block-editor/src/components/background-image-control/style.scss
index 971c877805..64dea516c7 100644
--- a/packages/block-editor/src/components/background-image-control/style.scss
+++ b/packages/block-editor/src/components/background-image-control/style.scss
@@ -1,13 +1,8 @@
 .block-editor-global-styles-background-panel__inspector-media-replace-container {
-	border: $border-width solid $gray-300;
 	border-radius: $radius-small;
 	// Full width. ToolsPanel lays out children in a grid.
 	grid-column: 1 / -1;
 
-	&.is-open {
-		background-color: $gray-100;
-	}
-
 	.block-editor-global-styles-background-panel__image-tools-panel-item {
 		flex-grow: 1;
 		border: 0;
@@ -56,6 +51,10 @@
 		&:hover {
 			color: var(--wp-admin-theme-color);
 		}
+
+		&:focus {
+			box-shadow: inset 0 0 0 var(--wp-admin-border-width-focus) var(--wp-admin-theme-color);
+		}
 	}
 
 	.block-editor-global-styles-background-panel__loading {
@@ -76,21 +75,16 @@
 	height: 100%;
 	width: 100%;
 	padding-left: $grid-unit-15;
-
-	&.is-open {
-		background-color: $gray-100;
-	}
-
-	&:focus {
-		box-shadow: 0 0 0 var(--wp-admin-border-width-focus) var(--wp-admin-theme-color);
-		outline: none;
-	}
 }
 
 .block-editor-global-styles-background-panel__dropdown-toggle {
 	cursor: pointer;
 	background: transparent;
 	border: none;
+
+	&.is-open {
+		background-color: $gray-100;
+	}
 }
 
 .block-editor-global-styles-background-panel__inspector-media-replace-title {
@@ -113,12 +107,33 @@
 .block-editor-global-styles-background-panel__inspector-image-indicator {
 	background-size: cover;
 	border-radius: $radius-small;
-
+	width: 20px;
+	height: 20px;
+	display: block;
+	position: relative;
+	// Show a thin outline in Windows high contrast mode, otherwise the button is invisible.
+	border: 1px solid transparent;
 	&.has-image {
-		background-image: var(--image-url);
+		background-image: var(--wp-admin-background-image-preview-url);
+	}
+	&.has-image::after {
+		background: unset;
 	}
 }
 
+.block-editor-global-styles-background-panel__inspector-image-indicator::after {
+	content: "";
+	position: absolute;
+	top: -1px;
+	left: -1px;
+	bottom: -1px;
+	right: -1px;
+	border-radius: $radius-small;
+	box-shadow: inset 0 0 0 $border-width rgba(0, 0, 0, 0.2);
+	background: #fff linear-gradient(-45deg, #0000 48%, #ddd 0, #ddd 52%, #0000 0);
+	box-sizing: inherit;
+}
+
 .block-editor-global-styles-background-panel__dropdown-content-wrapper {
 	min-width: 260px;
 	overflow-x: hidden;
@@ -138,6 +153,10 @@
 	.components-focal-point-picker::after {
 		content: none;
 	}
+
+	.block-editor-global-styles-background-panel__image-tools-panel-item {
+		padding: 0;
+	}
 }
 
 // Push control panel into the background when the media modal is open.
@@ -145,12 +164,6 @@
 	z-index: z-index(".block-editor-global-styles-background-panel__popover");
 }
 
-.block-editor-global-styles-background-panel__popover {
-	.block-editor-global-styles-background-panel__image-tools-panel-item {
-		padding: 0;
-	}
-}
-
 .block-editor-global-styles-background-panel__media-replace-popover {
 	.components-popover__content {
 		// width of block-editor-global-styles-background-panel__dropdown-content-wrapper minus padding.


```

</details>